### PR TITLE
Feature/3242 anon export task

### PR DESCRIPTION
**Pull request by philipkcl**
For [issue 3242](https://github.com/DOAJ/doajPM/issues/3242)

### CODE REVIEW

File: portality/background_helper.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+""" collections of wrapper function for helping you to create BackgroundTask
```
Comment (richard-jones) [inline portality/background_helper.py:1]: Can I suggest that we put this somewhere other than the root of the repo.  I propose in either `portality.lib` or `portality.tasks`.  I'm leaning towards the latter, but open to persuasion.

File: portality/tasks/anon_export.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                                            action=cls.__action__,
+                                            params=params)
+
+    @classmethod
+    def submit(cls, background_job):
+        background_job.save()
+        on_submit_anon_export.schedule(args=(background_job.id,), delay=10)
+
+
+@main_queue.periodic_task(schedule(AnonExportBackgroundTask.__action__))
```
Comment (richard-jones) [inline portality/tasks/anon_export.py:190]: as per other comment, suggest this goes on the long running queue

File: portality/tasks/anon_export.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+@write_required(script=True)
+def scheduled_anon_export():
+    background_helper.submit_by_bg_task_type(AnonExportBackgroundTask,
+                                             clean=app.config.get("TASKS_ANON_EXPORT_CLEAN", False),
+                                             limit=app.config.get("TASKS_ANON_EXPORT_LIMIT", None),
+                                             batch_size=app.config.get("TASKS_ANON_EXPORT_BATCH_SIZE", 100000))
+
+
+@main_queue.task()
```
Comment (richard-jones) [inline portality/tasks/anon_export.py:199]: as per other comment, suggest this goes on the long running queue

File: test.cfg [hunk 4]
```diff
@@ -52,10 +52,17 @@ HUEY_SCHEDULE = {
...
 # =======================
 # Account settings
 PUBLIC_REGISTER = True
 LOGIN_VIA_ACCOUNT_ID = True
+
+#########################################################
+# Background tasks --- anon export
+TASKS_ANON_EXPORT_CLEAN = False
+TASKS_ANON_EXPORT_LIMIT = None
+TASKS_ANON_EXPORT_BATCH_SIZE = 100000
```
Comment (richard-jones) [inline test.cfg:68]: Since these properties don't change the defaults in `settings.py` I think they can be omitted

**Review by richard-jones — CHANGES_REQUESTED** [review]
Very minor structural changes from me, otherwise this looks good to go.

@Steven-Eardley we'll want to run this up on test and give it a bit more of a workout there

File: doajtest/unit/test_tasks_anon_export.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    def test_execute(self):
+
+        # prepare test data
+        for _ in range(3):
+            BackgroundJob().save(blocking=True)
+        new_background_jobs = list(BackgroundJob.scroll())
+
+        for _ in range(2):
+            Account().save(blocking=True)
```
[the remark this answers is not in the document]
Comment (philipkcl) [reply doajtest/unit/test_tasks_anon_export.py:45]: so, elasticsearch process all save request by signal queue? I thought they save all record with mult-thread.

File: portality/background_helper.py (hunk 1 above)
Comment (philipkcl) [reply portality/background_helper.py:1]: `portality.lib` seem contain set of function that can be reuse / share to other project. but common pattern of background task could be only apply in DOAJ.
`portality.tasks` seem only contain file of `BackgroundTask`, I will move it to `portality.tasks.helpers`

**Comment by philipkcl** [comment]
@richard-jones update PR for feedback, please feel free to review

File: doajtest/unit/test_tasks_anon_export.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (richard-jones) [reply doajtest/unit/test_tasks_anon_export.py:43]: Hm, I don't know how it queues them :) All I can say is that we've never had a problem assuming that once the last to save is done, then everything before it is also done.  You can keep it or change it, I don't mind, it just makes the test slow is all.

**Review by richard-jones — APPROVED** [review]

**Comment by philipkcl** [comment]
@richard-jones and @Steven-Eardley  Some update for this branch. just let you known in case it already in testing stage.
just find out I should use `long_running` for `consumer_long_running.py`

**Comment by richard-jones** [comment]
Looks good to me.  This is now just queued for test

**Pull request merged**